pkg/s3select/select.go

```diff
@@ -175,41 +178,56 @@ func (reader *Input) processSelectReq(reqColNames []string, alias string, whereC
 	if limitOfRecords == 0 {
 		limitOfRecords = math.MaxInt64
 	}
-
 	for {
-		record := reader.ReadRecord()
-		reader.stats.BytesProcessed += processSize(record)
-		if record == nil {
+		individualRecord, errParsing := readIndividualRecord(inputType)
+		if errParsing != nil {
+			rowStruct := &Row{
+				err: errParsing,
+			}
+			myRow <- rowStruct
+			return
+		}
+		out, _ := json.Marshal(individualRecord)
+		inputType.getBytesProcessed(individualRecord)
+		if individualRecord == nil {
```

Comment:
this tyle is not clean, we shouldn't use `nil` to indicate that its CSV

Reply:
@harshavardhana  : Can you suggest a better approach. Will incorporate it.

Reply:
The problems with this approach are few basically json.Marshal errors will be ignored, we are combining both logic i.e json and CSV which are two different types of data into one framework. 

CSV cannot be parsed as JSON but there is a new overhead now to trying to marshal it anyways while failing to do so we send a CSV message.

The flow should be based on inputType, where we process these separately. i.e CSV shouldn't be processed as JSON at all.